xarray/core/common.py

```diff
@@ -1397,6 +1397,9 @@ def full_like(other, fill_value, dtype: DTypeLike = None):
     from .dataset import Dataset
     from .variable import Variable
 
+    if not np.isscalar(fill_value):
```

Comment:
could you use `xarray.core.utils.is_scalar` here instead? `np.isscalar` has some rather strange behavior, see the notes in its documentation for details: https://docs.scipy.org/doc/numpy/reference/generated/numpy.isscalar.html